### PR TITLE
refactor(account)!: rename send methods

### PR DESCRIPTION
**Pull request by Alex6323**
# Description of change

Renames some account methods related to sending as described in #586 .

## Links to any relevant issues

#586 

## Type of change

Choose a type of change, and delete any options that are not relevant.

- Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How the change has been tested

Unit tests

### CODE REVIEW

**Review by thibault-martinez — CHANGES_REQUESTED** [review]
Changelog

File: bindings/core/src/method_handler/account.rs [hunk 1]
```diff
@@ -277,14 +277,14 @@ pub(crate) async fn call_account_method_internal(account: &Account, method: Acco
         }
         AccountMethod::SendAmount { params, options } => {
```
Comment (Thoralf-M) [inline bindings/core/src/method_handler/account.rs]: Also rename the AccountMethods here?

Comment (Alex6323) [reply bindings/core/src/method_handler/account.rs]: I guess so :+1:

Comment (Alex6323) [reply bindings/core/src/method_handler/account.rs]: Done

**Review by Thoralf-M — DISMISSED** [review]

File: bindings/nodejs/lib/wallet/account.ts [hunk 2]
```diff
@@ -879,20 +879,20 @@ export class Account {
     }
 
     /**
-     * Prepare a send amount transaction, useful for offline signing.
+     * Prepare a send transaction, useful for offline signing.
```
Comment (kwek20) [inline bindings/nodejs/lib/wallet/account.ts]: prepare to send would be better now, old phrasing was weird anyway

File: bindings/python/examples/client/09_transaction.py [hunk 3]
```diff
@@ -15,11 +15,11 @@
...
 
-output = SendAmountParams(
+params = SendParams(
     address='rms1qzpf0tzpf8yqej5zyhjl9k3km7y6j0xjnxxh7m2g3jtj2z5grej67sl6l46',
     amount=1000000,
 )
 
 # Create and post a block with a transaction
-block = client.build_and_post_block(secret_manager, output=output)
+block = client.build_and_post_block(secret_manager, output=params)
```
Comment (thibault-martinez) [inline bindings/python/examples/client/09_transaction.py:24]: output=params is a bit weird

File: bindings/python/iota_sdk/client/client.py [hunk 4]
```diff
@@ -406,7 +406,7 @@ def build_and_post_block(self,
             End of the input range
         inputs : Array of Inputs
             Inputs to use
-        output : SendAmountParams
+        output : SendParams
```
Comment (thibault-martinez) [inline bindings/python/iota_sdk/client/client.py]: 🤔

File: sdk/examples/wallet/offline_signing/1_prepare_transaction.rs [hunk 5]
```diff
@@ -30,7 +30,7 @@ async fn main() -> Result<()> {
     // This example uses secrets in environment variables for simplicity which should not be done in production.
     dotenvy::dotenv().ok();
 
-    let outputs = [SendAmountParams::new(RECV_ADDRESS, SEND_AMOUNT)?];
+    let outputs = [SendParams::new(RECV_ADDRESS, SEND_AMOUNT)?];
```
Comment (thibault-martinez) [inline sdk/examples/wallet/offline_signing/1_prepare_transaction.rs]: rename?

File: sdk/src/wallet/account/operations/transaction/sign_transaction.rs [hunk 6]
```diff
@@ -24,7 +24,7 @@ impl<S: 'static + SecretManage> Account<S>
 where
     crate::wallet::Error: From<S::Error>,
 {
-    /// Function to sign a transaction essence
+    /// Account method to sign a transaction essence.
```
Comment (thibault-martinez) [inline sdk/src/wallet/account/operations/transaction/sign_transaction.rs]: I personally wouldn't put the type, it's a bit obvious and unusual. I'd just sat `Method to...`

File: bindings/python/examples/client/09_transaction.py (hunk 3 above)
Comment (Alex6323) [reply bindings/python/examples/client/09_transaction.py:24]: true ... had the same feeling.

File: sdk/examples/wallet/offline_signing/1_prepare_transaction.rs (hunk 5 above)
Comment (Alex6323) [reply sdk/examples/wallet/offline_signing/1_prepare_transaction.rs]: yes, missed this one

File: sdk/src/wallet/account/operations/transaction/sign_transaction.rs (hunk 6 above)
Comment (Alex6323) [reply sdk/src/wallet/account/operations/transaction/sign_transaction.rs]: Fine with me!

File: bindings/nodejs/lib/wallet/account.ts (hunk 2 above)
Comment (Alex6323) [reply bindings/nodejs/lib/wallet/account.ts]: Changed.

File: bindings/python/examples/client/09_transaction.py (hunk 3 above)
Comment (Alex6323) [reply bindings/python/examples/client/09_transaction.py:24]: I changed the function argument name to `params` as well.

File: bindings/python/iota_sdk/client/client.py (hunk 4 above)
Comment (Alex6323) [reply bindings/python/iota_sdk/client/client.py]: I changed the function argument name to `params` as well.

File: sdk/examples/wallet/offline_signing/1_prepare_transaction.rs (hunk 5 above)
Comment (Alex6323) [reply sdk/examples/wallet/offline_signing/1_prepare_transaction.rs]: Changed.

File: sdk/src/wallet/account/operations/transaction/sign_transaction.rs (hunk 6 above)
Comment (Alex6323) [reply sdk/src/wallet/account/operations/transaction/sign_transaction.rs]: I changed it to `Signs ...` as we sometimes do this anyway. So now it should be consistent.

File: bindings/python/examples/client/09_transaction.py [hunk 7]
```diff
@@ -15,11 +15,11 @@
...
 
-output = SendAmountParams(
+params = SendParams(
     address='rms1qzpf0tzpf8yqej5zyhjl9k3km7y6j0xjnxxh7m2g3jtj2z5grej67sl6l46',
     amount=1000000,
 )
 
 # Create and post a block with a transaction
-block = client.build_and_post_block(secret_manager, output=output)
+block = client.build_and_post_block(secret_manager, params=params)
```
Comment (thibault-martinez) [inline bindings/python/examples/client/09_transaction.py]: Is params=params needed if it has the correct name?

File: bindings/python/iota_sdk/client/client.py [hunk 8]
```diff
@@ -425,8 +425,8 @@ def build_and_post_block(self,
 
         options = {k: v for k, v in options.items() if v != None}
 
-        if 'output' in options:
-            options['output'] = options.pop('output').as_dict()
+        if 'params' in options:
+            options['output'] = options.pop('params').as_dict()
```
Comment (thibault-martinez) [inline bindings/python/iota_sdk/client/client.py]: `options['output']` ? Is it output in the message interface ?

Comment (Thoralf-M) [reply bindings/python/iota_sdk/client/client.py]: Yes https://github.com/iotaledger/iota-sdk/blob/85942111ac19a3998e945e2810563178178dab49/sdk/src/client/api/block_builder/mod.rs#L75
and this type is actually not the same as SendParams

Comment (thibault-martinez) [reply bindings/python/iota_sdk/client/client.py]: 🤔

Comment (Alex6323) [reply bindings/python/iota_sdk/client/client.py]: yeah, I wondered whether I should change that type as well.

File: bindings/python/examples/client/09_transaction.py (hunk 7 above)
Comment (Alex6323) [reply bindings/python/examples/client/09_transaction.py]: it's not needed I think, but it was like this before ... and I wasn't sure whether that's idiomatic Python.

Comment (Alex6323) [reply bindings/python/examples/client/09_transaction.py]: I am happy to remove it. @Thoralf-M wdyt?

File: bindings/python/iota_sdk/client/client.py (hunk 8 above)
Comment (Thoralf-M) [reply bindings/python/iota_sdk/client/client.py]: The other fields wouldn't be considered unless you also change that which I don't think we should do

File: bindings/python/examples/client/09_transaction.py (hunk 7 above)
Comment (Thoralf-M) [reply bindings/python/examples/client/09_transaction.py]: Sure remove

File: bindings/python/iota_sdk/client/client.py (hunk 8 above)
Comment (thibault-martinez) [reply bindings/python/iota_sdk/client/client.py]: Let's ignore for now, added an item to https://github.com/iotaledger/iota-sdk/issues/129

File: bindings/python/iota_sdk/client/client.py [hunk 9]
```diff
@@ -406,7 +406,7 @@ def build_and_post_block(self,
             End of the input range
         inputs : Array of Inputs
             Inputs to use
-        output : SendAmountParams
```
Comment (Thoralf-M) [inline bindings/python/iota_sdk/client/client.py:409]: Undo all the changes in this file and just do `output: Any  # TODO: https://github.com/iotaledger/iota-sdk/issues/129`

**Review by thibault-martinez — DISMISSED** [review]

**Review by thibault-martinez — DISMISSED** [review]

**Review by Thoralf-M — DISMISSED** [review]

**Review by Thoralf-M — APPROVED** [review]

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**